### PR TITLE
Use multi workspace client

### PR DESCRIPTION
**Pull request by razor-x**
- Load openapi schema from SDK or types
- Use SeamHttpMultiWorkspace

### CODE REVIEW

File: lib/get-seam.ts [hunk 1]
```diff
@@ -1,25 +1,37 @@
-import { SeamHttp } from "@seamapi/http/connect"
+import { SeamHttp, SeamHttpMultiWorkspace } from "@seamapi/http/connect"
 import { getConfigStore } from "./get-config-store"
 import { getServer } from "./get-server"
 
-export const getSeam = () => {
+export const getSeam = async (): Promise<SeamHttp> => {
```
Comment (razor-x) [inline lib/get-seam.ts:5]: This was being `await`ed in the code, so I made it async, but it could be sync as written.

File: lib/get-seam.ts [hunk 2]
```diff
@@ -1,25 +1,37 @@
...
   const config = getConfigStore()
 
   const token = config.get(`${getServer()}.pat`)
+
   const token_type = token.startsWith("seam_at")
```
Comment (seveibar) [inline lib/get-seam.ts:10]: this would be nice to have in the module fwiw

File: lib/get-seam.ts [hunk 3]
```diff
@@ -1,25 +1,31 @@
...
-    apiKey: token_type === "api-key" ? token : undefined,
-    // https://github.com/seamapi/javascript-http/issues/30
-    workspaceId: config.get("current_workspace_id"),
-  })
+
+  const workspaceId = config.get("current_workspace_id")
+
+  const options = { endpoint: getServer() }
+
+  if (token.startsWith("seam_at")) {
```
Comment (razor-x) [inline lib/get-seam.ts:14]: This logic is contained in the SDK but not exported. I wonder if we should export `isPersonalAccessToken` etc? Or `detectTokenFormat` there?

**Review by seveibar — APPROVED** [review]

File: lib/get-seam.ts (hunk 2 above)
Comment (razor-x) [reply lib/get-seam.ts]: Whoops, I think we have the same idea: https://github.com/seamapi/seam-cli/pull/46/files#r1434516504

File: lib/get-seam.ts (hunk 3 above)
Comment (seveibar) [reply lib/get-seam.ts:14]: Honestly I would also recommend SeamHttp.fromToken, I keep reimplementing this and it's error prone logic, it took me a couple releases and people reporting bugs to handle every case

Comment (razor-x) [reply lib/get-seam.ts:14]: I can support `SeamHttp.fromToken` but how to deal with the workspaceId or other "required" params that vary based on the token?

Comment (razor-x) [reply lib/get-seam.ts:14]: (Merging this for now but if we can get something upstream to handle it we can use it here)

**Comment by seveibar** [comment]
I'm getting this error locally

image

**Comment by seveibar** [comment]
(when using an api key)

**Comment by seveibar** [comment]
The issue is workspace selection is no longer working when you provide an api key to seam login

**Comment by seveibar** [comment]
Reverting for now

**Comment by seveibar** [comment]
fix incoming

**Comment by seveibar** [comment]
#55

**Comment by razor-x** [comment]
> The issue is workspace selection is no longer working when you provide an api key to seam login

How would you select a workspace when using an API key, isn't an API key bound to a single workspace?

**Pull request merged**